packages/amplify-category-analytics/src/analytics-resource-api.ts

```diff
@@ -9,34 +9,19 @@ import { addResource } from './provider-utils/awscloudformation/index';
 import { analyticsPush } from './commands/analytics';
 import { invokeAuthPush } from './plugin-client-api-auth';
 import { invokeNotificationsAPIGetAvailableChannelNames } from './plugin-client-api-notifications';
+import { pinpointHasInAppMessagingPolicy } from './utils/pinpoint-helper';
+import { getAnalyticsResources } from './utils/analytics-helper';
+import { analyticsMigrations } from './migrations';
 
 /**
  * Get all analytics resources. If resourceProviderService name is provided,
  * then only return resources matching the service.
  * @returns Array of resources in Analytics category (IAmplifyResource type)
  */
-export const analyticsPluginAPIGetResources = (resourceProviderServiceName?: string, context?: $TSContext): Array<IAnalyticsResource> => {
-  const resourceList: Array<IAnalyticsResource> = [];
-  const amplifyMeta = (context) ? context.exeInfo.amplifyMeta : stateManager.getMeta();
-  if (amplifyMeta?.[AmplifyCategories.ANALYTICS]) {
```

Comment:
method extracted in order to be reused in this category as well: packages/amplify-category-analytics/src/utils/analytics-helper.ts